### PR TITLE
Fix readLinearizedTempC() to use actual CJC temperature.

### PR DESCRIPTION
**Pull request by tgikal**
There are a few issues in the readLinearizedTempC() function related to correct use of the internal vs external temperature variables, changed the variables around so the calculations are correct, the issue would only appear as the measured temperature approaches the internal temperature.

Also changed the error handling to return the float 'NAN' instead of 0, which could result in erroneous information, if you were expecting to see 0°C.

### CODE REVIEW

**Comment by ladyada** [comment]
ooh hey yknow this is about to be archived and deprecated - we'll only be using our better-supported circuitpython library here
https://github.com/adafruit/Adafruit_CircuitPython_MAX31855
could you check if it has the fix in place and if not submit the PR there? we'll be closing this repo very soon!

**Comment by tgikal** [comment]
Thanks for the heads up, I'll look at the current repo.

**Comment by ladyada** [comment]
@tgikal hiya we're getting close to shutting off this repo - do you have the time to look at the https://github.com/adafruit/Adafruit_CircuitPython_MAX31855 and add this function there to test it? we could add it but we wouldnt really know how to fully test :)

**Comment by tgikal** [comment]
I would need to get components around to test it, I'm not 100% sure how the new circuit python libraries function, it looks like they have multiple components that need installed. My fix in this repository has been functioning well enough for my needs, the eight MAX31855 modules I've been using have held better than a +/-2C accuracy over a range of -65C to +200C every time I've checked them using calibrated equipment from Trescal and Vaisala.

Basically the linearized temperature focuses on correcting the thermocouple mV and the CJC mV to an established NIST table for type K thermocouples, since thermocouples do not follow a linear curve.

Where you would mostly see the correction taking effect is when the thermocouple and CJC are near the same temperature. So measuring 23C in a room at 23C may return 20C to 26C without the proper correction.

**Comment by ladyada** [comment]
if you're using it with raspberry pi, you can pip3 install all the supporting libraries, its waaay easier than this library!

**Comment by tgikal** [comment]
I think I have it written in, but I don't want to submit a pull request until I can test it, which currently I don't have access to any of the MAX31855 modules, since they are all in use (think I actually have 10 of them and can't spare 1 for testing, lol), I might get around to testing it early next week.

Added it in my fork:
https://github.com/tgikal/Adafruit_CircuitPython_MAX31855

**Comment by ladyada** [comment]
ok cool, any PR would be great, whats nice is that with circuitpython this code will work with any/all linux boards like nanopi, odroid, etc...will be way more useful to the community of SBC hackers :)

**Comment by tgikal** [comment]
I'm having trouble figuring out how to create a software SPI connection:
`spi = bitbangio.SPI(board.D15, MISO=board.D14)`

isn't complete it appears... is there any way to use a software SPI connection with the new CircuitPython modules? Otherwise I can see one advantage of this current module already ;)

**Comment by ladyada** [comment]
at this time, please use hardware SPI - we dont have bitbangio yet!

**Comment by tgikal** [comment]
So my problem is I'm using the adafruit 2423 screen as well and never even tried to use the hardware SPI because the screen is already using it, any idea whether the screen would have troubles sharing the SPI lines?

**Comment by ladyada** [comment]
hmm is it possible for you to test the PR with just text output?

**Comment by ladyada** [comment]
hiya this library is being deprecated, and replaced with our shiny new circuitpython library, which has this function ported over
we are archiving this repo
please check out https://github.com/adafruit/Adafruit_CircuitPython_MAX31855/blob/master/adafruit_max31855.py#L95
and if it is still amiss please open an issue in that repo!
